lib/PHPExif/Mapper/Native.php

```diff
@@ -246,7 +247,8 @@ public function mapRawData(array $data) : array
                     break;
                 case self::GPSLONGITUDE:
                     $GPSLongitudeRef = '';
-                    if (array_key_exists('GPSLongitudeRef', $data) && $data['GPSLongitudeRef'][0] !== '') {
+                    if (array_key_exists('GPSLongitudeRef', $data)
+                        && $data['GPSLongitudeRef'] !== null && $data['GPSLongitudeRef'][0] !== '') {
                         $GPSLongitudeRef = $data['GPSLongitudeRef'][0];
                     }
```

Comment:
To me this condition appears even more robust than the other ones in `Exiftool` and `ImageMagick`. Here we additionally test for `!== null`. But we do not set `N` (or `E`) as the default when the string is empty as we do for the other two extractors.

Why this asymmetry?

I would have expected the best of both, i.e. the more robust condition and setting a default.

Reply:
The reason is that with the native extractor we actually _do_ get `null` with the image file in question whereas with the other two we get empty strings instead.

But you are right, it would've been better to code it more defensively. I'll unify that across the mappers.